src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -232,6 +189,112 @@ impl<'a> Visitor<'a> for CollectCustomDerives<'a> {
             };
             self.handler.span_err(item.span, msg);
         }
+    }
+
+    fn collect_attr_proc_macro(&mut self, item: &'a ast::Item, attr: &'a ast::Attribute) {
```

Comment:
I think this should be merged with `collect_custom_derive` to avoid duplicating code (optional)